src/commands/Moderation/multiban.ts

```diff
@@ -28,93 +28,112 @@ export default class extends Command {
 	}
 
 	public async run(context: CommandContext, args: string[]): Promise<void | Message> {
-		const targets: RegExpMatchArray = args.slice(0).join(" ").match(UserMentionAndID) ?? [];
-		let reason: string = args.slice(targets.length).join(" ").replace(UserMentionAndID, "");
+		const potentialTargets: RegExpMatchArray = <RegExpMatchArray>args.slice(0).join(" ").match(UserMentionAndID);
+		let validTargets: (GuildMember | User)[] = [];
+		let invalidTargets: number = 0;
+		let fullList: string[] = [];
+		let reason: string = args.slice(potentialTargets?.length).join(" ").replace(UserMentionAndID, "");
 
 		if (reason === "") reason = await this.client.bulbutils.translate("global_no_reason", context.guild?.id, {});
-		let fullList: string = "";
+		if (!potentialTargets) return context.channel.send(await this.client.bulbutils.translate("action_multi_no_valid_targets", context.guild?.id, {}));
 
-		if (targets.length <= 1) {
+		for (const potentialTarget of potentialTargets) {
+			const t = potentialTarget.replace(NonDigits, "");
+			if (!t.length) continue;
+
+			let target: User | GuildMember | null = t ? <GuildMember>await context.guild?.members.fetch(t).catch(() => null) : null;
+
+			if (!target) {
+				try {
+					target = await this.client.users.fetch(t);
+				} catch (error) {
+					invalidTargets++;
+					continue;
+				}
+			}
+
+			if (target instanceof GuildMember)
+				if (await this.client.bulbutils.resolveUserHandle(context, await this.client.bulbutils.checkUser(context, target), target.user)) {
+					invalidTargets++;
+					continue;
+				}
+
+			if (
+				validTargets.find(user => {
+					if (user instanceof GuildMember && target instanceof GuildMember) return user.user.id === target.user.id;
+					else if (user instanceof User && target instanceof User) return user.id === target?.id;
+					else return false;
+				})
+			) continue;
+				validTargets.push(target);
+		}
+
+		if (validTargets.length === 1) {
 			await context.channel.send(
 				await this.client.bulbutils.translate("action_multi_less_than_2", context.guild?.id, {
 					action: await this.client.bulbutils.translate("action_multi_types.ban", context.guild?.id, {}),
 				}),
 			);
-			return await this.client.commands.get("ban")!.run(context, args);
+			return this.client.commands.get("ban")!.run(context, [validTargets[0].id, ...reason.split(" ")]);
 		}
 
-		context.channel.send(await this.client.bulbutils.translate("global_loading", context.guild?.id, {})).then(msg => {
-			setTimeout(() => msg.delete(), (args.length - 0.5) * massCommandSleep);
-		});
+		const msg: Message = await context.channel.send(await this.client.bulbutils.translate("global_loading", context.guild?.id, {}));
 
-		for (let i = 0; i < targets.length; i++) {
-			if (targets[i] === undefined) continue;
+		for (const target of validTargets) {
```

Comment:
Currently this command does not check whether a target is already banned. This is probably not ideal